challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -10506,6 +10506,1096 @@
+        },
+        {
+          "text":
+            "'<code>non_continuous_subsequences('+JSON.stringify(tests[3])+')</code> should return <code>'+JSON.stringify(results[3])+'</code>.'",
+          "testString":
+            "assert.deepEqual(non_continuous_subsequences(tests[3]),results[3],'<code>non_continuous_subsequences('+JSON.stringify(tests[3])+')</code> should return <code>'+JSON.stringify(results[3])+'</code>.');"
+        },
+        {
+          "text":
+            "'<code>non_continuous_subsequences('+JSON.stringify(tests[4])+')</code> should return <code>'+JSON.stringify(results[4])+'</code>.'",
+          "testString":
+            "assert.deepEqual(non_continuous_subsequences(tests[4]),results[4],'<code>non_continuous_subsequences('+JSON.stringify(tests[4])+')</code> should return <code>'+JSON.stringify(results[4])+'</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f35",
+      "challengeType": 5,
+      "releasedOn": "July 28, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": [
+            "function non_continuous_subsequences (ary) {",
+            "  // Good luck!",
+            "}"
+          ],
+          "head": [],
+          "tail": [
+            "var tests=[",
+            "[1,2,3],",
+            "[1,2,3,4],",
+            "[1,5,9],",
+            "[1,2,3,4,5],",
+            "[5,18,14]]",
+            "var results=[",
+            "[[1,3]],",
+            "[[1,3],[1,4],[2,4],[1,2,4],[1,3,4]],",
+            "[[1,5],[1,9],[5,9],[1,5,9]],",
+            "[[1,3],[1,4],[2,4],[1,2,4],[1,3,4],[1,5],[2,5],[1,2,5],[3,5],[1,3,5],[2,3,5],[1,2,3,5],[1,4,5],[2,4,5],[1,2,4,5],[1,3,4,5]],",
+            "[[5,18],[5,14],[18,14],[5,18,14]]]"
+          ]
+        }
+      }
+    },
+    {
+      "title": "Nonoblock",
+      "description": [
+        "Nonoblock is a chip off the old <a href=\"http://rosettacode.org/wiki/Nonogram solver\">Nonogram</a> puzzle.",
+        "Given a row of five cells and a block of two cells followed by a block of 1 cell - in that order, the example could be shown as:",
+        "[0,0,0,0,0] # 5 cells and [2, 1] blocks",
+        "And would expand to the following 3 possible rows of block positions:",
+        "[1,1,0,1,0]",
+        "[1,1,0,0,1]",
+        "[0,1,1,0,1]",
+        "Note how the sets of blocks are always separated by a 0.",
+        "Write a function that takes a string and a number as parameters. The string contains digits and gives the size of the blocks. For example: \"21\" = [2,1] blocks. The second parameter is the length/number of cells. The function should return an array with all possible solutions. Each element should be a bit array (array consisting of 0s and 1s) that represents a solution."
```

Comment:
`"Write a function that takes a string and a number as parameters. The string contains digits and gives the size of the blocks. For example: \"21\" = [2,1] blocks. The second parameter is the length/number of cells. The function should return an array with all possible solutions. Each element should be a bit array (an array consisting of 0s and 1s) that represents a solution."`